### PR TITLE
Add `.`, `-`, and `_` to set of unescaped characters in `urlencode`

### PR DESCRIPTION
**Pull request by lucky**
I ran `make test` and then `cargo insta review` to verify the changes. I'm not sure if there's anything else that's required. (See below for a note on a line that changed and I'm unsure why.)

### CODE REVIEW

File: minijinja/tests/snapshots/test_templates__vm@filters.txt.snap [hunk 1]
```diff
@@ -1,9 +1,7 @@
 ---
 source: minijinja/tests/test_templates.rs
-assertion_line: 44
```
Comment (lucky) [inline minijinja/tests/snapshots/test_templates__vm@filters.txt.snap:3]: I don't know why this changed

**Comment by lucky** [comment]
@mitsuhiko I believe this is ready for review. I'm also interested in creating a PR for the README to add a section on how to develop for this repository. Any feedback here is helpful, thanks!

File: minijinja/tests/snapshots/test_templates__vm@filters.txt.snap (hunk 1 above)
Comment (mitsuhiko) [reply minijinja/tests/snapshots/test_templates__vm@filters.txt.snap:3]: I think your version of `cargo-insta` is probably too old.

Comment (mitsuhiko) [reply minijinja/tests/snapshots/test_templates__vm@filters.txt.snap:3]: Actually in this case it's the other way round. Snapshots were made on an older insta version.

**Comment by mitsuhiko** [comment]
Thank you for this. Happy to accept a PR for contributing. Ideally put the main contents into a `CONTRIBUTING.md` file. I made one for `insta` which could be a point of reference: https://github.com/mitsuhiko/insta/blob/master/CONTRIBUTING.md

**Pull request merged**